poi-ooxml/src/main/java/org/apache/poi/openxml4j/opc/ZipPackage.java

```diff
@@ -511,7 +524,8 @@ protected void closeImpl() throws IOException {
      * @return Path to a temporary file
      */
     private synchronized Path generateTempPath() throws IOException {
-        return File.createTempFile("OpenXML4J" + System.nanoTime(), ".tmp").toPath();
+        File tempDirectoryFile = tempDirectory != null ? tempDirectory.toFile() : null;
+        return File.createTempFile("OpenXML4J" + System.nanoTime(), ".tmp", tempDirectoryFile).toPath();
```

Comment:
Default behavior for this API is to write to the default temp directory when the given path is null.